### PR TITLE
specify zip pattern instead of zip length in config

### PR DESCRIPTION
**Pull request by fossecode**

### CODE REVIEW

**Comment by tkrunning** [comment]
Maltese config:
`{
  "BASE_URL": "coronastatusmt.com",
  "LANGUAGE": "en-MT",
  "LOCALE": "en-MT",
  "COUNTRY_CODE": "mt",
  "COUNTRY": "Malta",
  "MAP_CENTER": "14.3582, 35.9401",
  "MAP_ZOOM": 10,
  "MAP_MAX_ZOOM": 14,
  "TWITTER": "coronastatusMT",
  "ZIP_PATTERN": "[A-Za-z]{2,3}",
  "ZIP_PLACEHOLDER": "Example: VLT (no numbers)",
  "ZIP_GUIDE": true,
  "REDIRECT_TO_GOVERNMENT": false,
  "PASSCODE_LENGTH": 3,
  "DB_PATH": "./covid_db"
}
`
It's probably best if you double check my regex ;)

**Comment by fossecode** [comment]
> { "BASE_URL": "coronastatusmt.com", "LANGUAGE": "en-MT", "LOCALE": "en-MT", "COUNTRY_CODE": "mt", "COUNTRY": "Malta", "MAP_CENTER": "14.3582, 35.9401", "MAP_ZOOM": 10, "MAP_MAX_ZOOM": 14, "TWITTER": "coronastatusMT", "ZIP_PATTERN": "[A-Za-z]{2,3}", "ZIP_PLACEHOLDER": "Example: VLT (no numbers)", "ZIP_GUIDE": true, "REDIRECT_TO_GOVERNMENT": false, "PASSCODE_LENGTH": 3, "DB_PATH": "./covid_db" }

Thanks, looks good! So zip code can only be letters?

File: app/config.ts [hunk 1]
```diff
@@ -23,7 +24,7 @@ const fallbackConfig: Config = {
   MAP_ZOOM: parseInt(process.env.MAP_ZOOM || '4', 10),
   MAP_MAX_ZOOM: parseInt(process.env.MAP_MAX_ZOOM || '13', 10),
   TWITTER: process.env.TWITTER || 'coronastatusNO',
-  ZIP_LENGTH: parseInt(process.env.ZIP_LENGTH || '4', 10),
+  ZIP_PATTERN: process.env.ZIP_PATTERN || '[A-Za-z0-9]{4}',
```
Comment (michaelmcmillan) [inline app/config.ts]: Should we default to allow letters? I guess this won't really be a problem – but could be helpful with a comment on why we would fallback to this (as opposed to just `[0-9]{4}` :)

Comment (fossecode) [reply app/config.ts]: I changed it to `[A-Za-z0-9]{2,10}` 😅 If someone forgets to set the pattern, I guess it is better to have very little validation.

**Comment by tkrunning** [comment]
> Thanks, looks good! So zip code can only be letters?

Yes, at least the ones that were provided from the service you linked to. In reality they are three letters (for the locality), then four digits. But due to the small size of Malta, if you include the full post code I think you in many cases could identify the individual houses, which wouldn't preserve the privacy of the users.

So yes, for our purposes I think it makes sense to only use the three initial letters (which match what was provided).

**Review by michaelmcmillan — APPROVED** [review]

**Pull request merged**